src/components/selections/select.js

```diff
@@ -809,8 +809,8 @@ function determineSearchTraces(gd, xAxes, yAxes, subplot) {
             var sankeyInfo = createSearchInfo(trace._module, cd, xAxes[0], yAxes[0]);
             searchTraces.push(sankeyInfo);
         } else {
-            if(xAxisIds.indexOf(trace.xaxis) === -1) continue;
-            if(yAxisIds.indexOf(trace.yaxis) === -1) continue;
+            if(xAxisIds.indexOf(trace.xaxis) === -1 && (!trace._xA || !trace._xA.overlaying)) continue;
+            if(yAxisIds.indexOf(trace.yaxis) === -1 && (!trace._yA || !trace._yA.overlaying)) continue;
```

Comment:
Very interesting proposed fix.
It looks like we also select traces in other subplots that are not overlay with these axes.
If that's the case we shouldn't include them.

Reply:
I have checked all the other test cases with subplots and the selection seems to be working fine. 

The only cases I've found where there are some crossover with selection/deselection are with:
- `plot_types`
- `plot_types_grid_dash`
- `box_plot_jitter_edge_cases`
- `mapbox_choropleth-multiple`

And for these the issues happen with and without my changes.

Reply:
@Farkites Thanks very much for your tests.
Here we need to know if we have two subplots i.e. (xy + xy2 ) and (x2y3 + x2,y4) and the user reselects over the first one; then are the traces in the second subplot (x2y3) are also re-selected before and after this change?

Reply:
@archmoj Yes, in that scenario the traces in the second subplot (x2y3) are re-selected before and after the changes. You can check this here:
- Before changes: https://codepen.io/Mario-Rodr-guez-Ib-ez/pen/ExMEbdK
- After changes: https://codepen.io/Mario-Rodr-guez-Ib-ez/pen/QWoQoao